### PR TITLE
Happi settings

### PR DESCRIPTION
**Pull request by ronpandolfi**

### CODE REVIEW

File: xicam/Acquire/controllers/areadetector.py [hunk 1]
```diff
@@ -49,7 +49,10 @@ def __init__(self, device, maxfps=1):
         self.layout().addWidget(self.imageview)
         self.layout().addWidget(self.passive)
 
-        pvname = device.pvname
+        pvname = device.prefix
+      
+        from ..devices.fastccd import FastCCD
+        device._device_obj = device.device_obj = FastCCD(prefix=pvname, name=device.name)
```
Comment (ronpandolfi) [inline xicam/Acquire/controllers/areadetector.py]: Remove hardcoded device class

**Comment by ihumphrey** [comment]
- [ ] Remove Device class 
- [ ] Make sure we use the controller abstraction classes
- [ ] Add happi mongo backend discovery

File: xicam/Acquire/controlwidgets/deviceview.py [hunk 2]
```diff
@@ -33,8 +33,14 @@ def __init__(self):
     #         except BadPVName:
     #             msg.logMessage(f'PV named "{pvname}" is invalid.', msg.WARNING)
 
-    def selectionChanged(self, *args, **kwargs):
-        indexes = self.selectedIndexes()
-        if indexes and indexes[0].isValid():
-            item = self._model.itemFromIndex(indexes[0])
-            self.sigShowControl.emit(item.widget)
+    # def selectionChanged(self, *args, **kwargs):
```
Comment (ronpandolfi) [inline xicam/Acquire/controlwidgets/deviceview.py]: remove comment block

File: xicam/Acquire/controlwidgets/typhosmotorcontroller.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from xicam.plugins import ControllerPlugin
+from typhos import TyphosDeviceDisplay
+import typhos
+
+
+class TyphosMotorController(TyphosDeviceDisplay, ControllerPlugin):
```
Comment (ronpandolfi) [inline xicam/Acquire/controlwidgets/typhosmotorcontroller.py]: rename -> TyphosController
check for controller_class field in happi config to decide which controller to use for display

**Comment by ronpandolfi** [comment]
@hhslepicka @klauer @ZLLentz Finishing up the happi plugin for Xi-CAM. 
image

**Comment by klauer** [comment]
👏

**Comment by birdzix** [comment]
approved :heavy_check_mark:

**Pull request merged**